var/spack/repos/builtin/packages/dd4hep/package.py

```diff
@@ -141,6 +141,8 @@ class Dd4hep(CMakePackage):
     depends_on("boost +system +filesystem", when="%gcc@:7")
     depends_on("root @6.08: +gdml +math +python")
     depends_on("root @6.08: +gdml +math +python +x +opengl", when="+ddeve")
+    depends_on("root +webgui", when="+ddeve ^root@6.28:")
+    depends_on("root @6.08:6.27", when="@:1.23 +ddeve")
```

Comment:
Cf. https://github.com/spack/spack/pull/35624#issuecomment-1440756500:
```suggestion
    with when("+ddeve"):
        depends_on("root @6.08: +x +opengl")
        depends_on("root +webgui", when="^root@6.28:")
        depends_on("root @:6.27", when="@:1.23")
```